### PR TITLE
Gracefully handle Bitmap Compression Failure

### LINKED ISSUE #5513: [2.9 ACRA] crash adding bitmap to card
https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/47c1cab0-f57c-4162-aa09-f09d416bd11b

```
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=2, result=-1, data=Intent { act=inline-data (has extras) }} to activity {com.ichi2.anki/com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.graphics.Bitmap.compress(android.graphics.Bitmap$CompressFormat, int, java.io.OutputStream)' on a null object reference
at android.app.ActivityThread.deliverResults(ActivityThread.java:4382)
at android.app.ActivityThread.handleSendResult(ActivityThread.java:4426)
at android.app.ActivityThread.-wrap20(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1685)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6626)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.graphics.Bitmap.compress(android.graphics.Bitmap$CompressFormat, int, java.io.OutputStream)' on a null object reference
at com.ichi2.anki.multimediacard.fields.BasicImageFieldController.rotateAndCompress(BasicImageFieldController.java:6)
at com.ichi2.anki.multimediacard.fields.BasicImageFieldController.onActivityResult(BasicImageFieldController.java:9)
at com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity.onActivityResult(MultimediaEditFieldActivity.java:3)
at android.app.Activity.dispatchActivityResult(Activity.java:7305)
at android.app.ActivityThread.deliverResults(ActivityThread.java:4378)
... 9 more
```

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
Sometimes, the `rotateAndCompress` method could fail, causing a crash.

## Fixes
Fixes #5513

## Approach
Instead, we return the uncompressed image, and display both the file size and a warning that some images may require compression.

We do this, rather than displaying an error, as it's easier for a user to take action.

We unconditionally display the file size as it's a useful attribute.

## How Has This Been Tested?

On my Android, both with a sample failure via the debugger and with a normal image via camera.

Note: Compression increased the image size.

image
image

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/res/values/02-strings.xml [hunk 1]
```diff
@@ -241,4 +241,5 @@
     <string name="export_v2_forbidden">Please switch to the V1 scheduler before exporting a single deck with scheduling information.</string>
 
     <string name="activity_start_failed">The system does not have an app installed that can perform this action.</string>
+    <string name="multimedia_editor_image_compression_failed"><![CDATA[Camera images may be large. You may wish to compress & resize images in the media directory]]></string>
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/values/02-strings.xml]: This requires some research, I'm not sure how the crowdin platform will handle the CDATA wrapping. Is it possible to do the CDATA outside the string resource?

Comment (david-allison-1) [reply AnkiDroid/src/main/res/values/02-strings.xml]: Existing translations already contain CDATA.

Easier for me to just remove the &

Comment (mikehardy) [reply AnkiDroid/src/main/res/values/02-strings.xml]: I hadn't checked - just noted it - if it follows existing translations -> resolved / works for me (either way)

**Comment by mikehardy** [comment]
Just waiting on CI for this one, it's being a real pain which will be mitigated for things based off master from the last few minutes with #5853 until then I am only succeeding for the macOS builds about 10% of the time 😞 but they'll all get through eventually

**Comment by mikehardy** [comment]
released in 2.9.6beta4

**Pull request merged**